From 2ac891fbfb92045c086b39f35b65d56dcf10f850 Mon Sep 17 00:00:00 2001
From: Diego Rondini <diego.rondini@kynetics.com>
Date: Wed, 21 Jun 2017 08:40:43 +0000
Subject: [PATCH] Log save_state

---
 core/state.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/state.c b/core/state.c
index 45e461b..239897e 100644
--- a/core/state.c
+++ b/core/state.c
@@ -29,7 +29,9 @@
 static server_op_res_t do_save_state(char *key, char* value)
 {
 	CHECK_STATE_VAR(key);
-	return bootloader_env_set(key, value) == 0 ? SERVER_OK : SERVER_EERR;
+	int ret = bootloader_env_set(key, value);
+	INFO("Saved value '%s' to key '%s' with return value '%d'.\n", value, key, ret);
+	return ret == 0 ? SERVER_OK : SERVER_EERR;
 }
 
 server_op_res_t save_state(char *key, update_state_t value)
