From f709d6dec703654c916a60c6ac69b00feb154311 Mon Sep 17 00:00:00 2001
From: Diego Rondini <diego.rondini@kynetics.com>
Date: Wed, 24 Apr 2019 09:21:58 +0200
Subject: [PATCH] Disable debug messages

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 118fc32..6192981 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,7 @@ project (libubootenv C)
 set(VERSION	"0.1")
 SET(SOVERSION "${VERSION}")
 add_definitions(-DVERSION="${VERSION}")
+add_definitions(-DNDEBUG=1)
 
 #set(CMAKE_C_FLAGS_DEBUG "-g")
 include_directories ("${PROJECT_SOURCE_DIR}/src")
-- 
2.17.2

